### PR TITLE
Implemented NodeJS compatibility test.

### PR DESCRIPTION
**Pull request by RazzM13**
Enhances #111.

### CODE REVIEW

**Comment by RazzM13** [comment]
I believe #127 should be merged first.

**Comment by martysweet** [comment]
@RazzM13 See comments on #111

**Comment by martysweet** [comment]
Can the checking scripts be moved to a new folder? Perhaps `tools/nodejs_compatibility_checker`

**Comment by RazzM13** [comment]
Addressed the above mentioned issue, the last one from #111 and I also took the liberty of guarding against a false-negative scenario. Ready for review :)

File: tools/nodejs_compatibility_checker/test_compatibility.sh [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+[ ! -e ~/.envirius/nv ] && echo 'Error: envirius is not installed!' && exit 1
+[ ! -e nodes_available.log ] && echo 'Error: no environments available!' && exit 1
+
+export TERM="dumb"
+source ~/.envirius/nv
+node_versions=$(cat nodes_available.log)
+
+# Run the tests on all available versions
+for version in ${node_versions}; do
+echo "Testing node-${version}!";
```
Comment (martysweet) [inline tools/nodejs_compatibility_checker/test_compatibility.sh]: Can this for block be indented?

File: tools/nodejs_compatibility_checker/test_compatibility_setup.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+# Install envirius
+git clone --depth 1 https://github.com/ekalinin/envirius.git && \
+cd envirius && \
+make install && \
+cd .. && \
+source ~/.envirius/nv
+
+if [ $? -ne 0 ]; then
+echo "Unable to install envirius!"
```
Comment (martysweet) [inline tools/nodejs_compatibility_checker/test_compatibility_setup.sh]: Indent

File: tools/nodejs_compatibility_checker/test_compatibility_setup.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  for lts in 4 6 8 9; do
+    echo "${node_versions}" | grep -E "^${lts}" | tail -1
+  done
+  )
+  node_versions=${lts_versions}
+fi
+
+# Dump available versions to file
+for version in ${node_versions}; do
+echo ${version} >> nodes_available.log
```
Comment (martysweet) [inline tools/nodejs_compatibility_checker/test_compatibility_setup.sh]: Indent

File: tools/nodejs_compatibility_checker/test_compatibility_setup.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+node_versions=$(nv ls-versions --node-prebuilt | sed 's/\s/\n/g' | grep -E '^[1-9]')
+if [ -z "${node_versions}" ]; then
+  echo "Unable to determine node versions!"
+  exit 1
+fi
+
+# Quick check
+if [ "$1" == "--quick" ]; then
+  lts_versions=$(
+  for lts in 4 6 8 9; do
```
Comment (martysweet) [inline tools/nodejs_compatibility_checker/test_compatibility_setup.sh]: Indent

File: tools/nodejs_compatibility_checker/test_compatibility_cache.sh [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+[ ! -e ~/.envirius/nv ] && echo 'Error: envirius is not installed!' && exit 1
+[ ! -e nodes_available.log ] && echo 'Error: no environments available!' && exit 1
+
+export TERM="dumb"
+source ~/.envirius/nv
+node_versions=$(cat nodes_available.log)
+
+# Cache NodeJS environments
+for version in ${node_versions}; do
+echo "Adding NodeJS ${version} into environment cache!" && \
```
Comment (martysweet) [inline tools/nodejs_compatibility_checker/test_compatibility_cache.sh]: Indent

File: tools/nodejs_compatibility_checker/test_compatibility.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  exit 0
+fi
+
+echo "Build is incompatible with the following node versions:"
+echo "${failed}"
+for version in ${failed}; do
+  echo "***Started test-run for ${version}***";
+  cat "test_node_${version}.log";
+  echo "***Finished test-run for ${version}***";
+done
```
Comment (martysweet) [inline tools/nodejs_compatibility_checker/test_compatibility.sh]: Nice touch

File: .circleci/config.yml [hunk 7]
```diff
@@ -28,6 +28,21 @@ jobs:
       - run:
           name: test
           command: npm test
+      - run:
```
Comment (martysweet) [inline .circleci/config.yml]: 22 seconds, awesome!

**Comment by RazzM13** [comment]
I think everything should be fine now ;)

File: tools/nodejs_compatibility_checker/test_compatibility_setup.sh [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
```
Comment (martysweet) [inline tools/nodejs_compatibility_checker/test_compatibility_setup.sh]:
```
lts_version=$(
    for lts in 4....

    done
)
```

**Comment by RazzM13** [comment]
I'm really bad with indentation :)) Hopefully it's ok now.

**Pull request merged**